src/main/java/com/google/api/codegen/transformer/SampleTransformer.java

```diff
@@ -211,34 +212,45 @@ public void generateSamples(
                     .build());
       }
 
-      for (ValueSetAndTags valueSet : matchingValueSets) {
-        InitCodeView initCodeView =
-            sampleGenerator.generate(
-                initContext != null
-                    ? initContext
-                    : createInitCodeContext(
-                        context,
-                        fieldConfigs,
-                        initCodeOutputType,
-                        valueSet.values().getParametersList(),
-                        valueSet.values().getSampleArgsList()));
-        List<OutputSpec> outputs = valueSet.values().getOnSuccessList();
+      for (ValueSetAndTags setAndTag : matchingValueSets) {
+        // Don't overwrite the initContext in outer scope.
+        InitCodeContext thisContext = initContext;
+        SampleValueSet valueSet = setAndTag.values();
+        if (thisContext == null) {
+          // TODO(pongad): Check that the attributes don't overlap.
+          // Probably do this after we implement the second kind of attributes;
+          // with only one implemented, I'm not sure what's the best way to do this yet.
```

Comment:
I don't think this depends on the types of attributes, but rather whether a part of the request object is covered by more than one: ie specifying a `SampleInitAttribute` for `request.list[0]` and another one for `request.list` should be an error.

OK to leave this as a TODO for now, but we should address it before we release.